### PR TITLE
feat: swapper avatar icons in trade quotes

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

icons for swappers in trade quotes!

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

makes @willyogo happy

## Risk

minimal - view layer change only

## Testing

we should see icons for the respective swappers per the screenshot - see the files tab for the icons added 👆👉

### Engineering

👆

### Operations

👆

## Screenshots (if applicable)

image

### CODE REVIEW

File: src/components/Trade/Components/TradeQuotes/TradeQuote.tsx [hunk 1]
```diff
@@ -161,6 +159,27 @@ export const TradeQuoteLoaded: React.FC<TradeQuoteLoadedProps> = ({
...
+        return CowIcon
+      case SwapperType.ZrxAvalanche:
+      case SwapperType.ZrxBnbSmartChain:
+      case SwapperType.ZrxEthereum:
+      case SwapperType.ZrxOptimism:
+        return ZrxIcon
+      case SwapperType.Thorchain:
+        return THORChainIcon
+      case SwapperType.Test:
+      default:
```
Comment (0xdef1cafe) [inline src/components/Trade/Components/TradeQuotes/TradeQuote.tsx:178]: side note - we should add a lint rule to allow exhaustive switches without default cases

File: src/components/Trade/Components/TradeQuotes/TradeQuote.tsx [hunk 2]
```diff
@@ -161,6 +159,27 @@ export const TradeQuoteLoaded: React.FC<TradeQuoteLoadedProps> = ({
...
+        return ''
+    }
+  }, [swapperWithMetadata])
```
Comment (0xApotheosis) [inline src/components/Trade/Components/TradeQuotes/TradeQuote.tsx:181]: If we want to ensure that all possible `SwapperType`s are always covered:

```suggestion
  const protocolIcon = useMemo(() => {
    const swapperType = swapperWithMetadata.swapper.getType()
    switch (swapperType) {
      case SwapperType.Osmosis:
        return OsmosisIcon
      case SwapperType.LIFI:
        return LiFiIcon
      case SwapperType.CowSwap:
        return CowIcon
      case SwapperType.ZrxAvalanche:
      case SwapperType.ZrxBnbSmartChain:
      case SwapperType.ZrxEthereum:
      case SwapperType.ZrxOptimism:
        return ZrxIcon
      case SwapperType.Thorchain:
        return THORChainIcon
      case SwapperType.Test:
        return ''
      default:
        assertUnreachable(swapperType)
    }
  }, [swapperWithMetadata])
```

**Review by 0xApotheosis — APPROVED** [review]
Functionally solid, one non-blocking comment.

🎨

File: src/components/Trade/Components/TradeQuotes/TradeQuote.tsx [hunk 3]
```diff
@@ -161,6 +159,27 @@ export const TradeQuoteLoaded: React.FC<TradeQuoteLoadedProps> = ({
     return borderColor
   })()
 
+  const protocolIcon = useMemo(() => {
+    switch (swapperWithMetadata.swapper.getType()) {
```
Comment (gomesalexandre) [inline src/components/Trade/Components/TradeQuotes/TradeQuote.tsx:163]: Actually agree with @0xApotheosis not to call this inline within the switch, even though this shouldn't throw, this is JS so always better be safe

**Review by gomesalexandre — APPROVED** [review]
pingz

image
image
image

**Review by gomesalexandre — APPROVED** [review]

**Pull request merged**